woocommerce/woocommerce-ios
# Dynamic Dashboard: Display unavailable analytics view when logged in without WPCom

Part of: #13157 


## Description

Previously, we were displaying the analytics unavailable view only when logged in with WPCom because only `DotComError.noRestRoute` error is checked. When logged in without WPCom, network error code 404 is returned instead.

This PR adds improvements for 4 cards: performance, top performers, stock, and most active coupon cards. Changes include:
- Created a reusable view `UnavailableAnalyticsView` to be displayed on all 4 cards when the analytics module is unavailable.
- Updated the logic for all 4 cards to check for both `DotcomError.noRestRoute` and `NetworkError.notFound` to display the unavailable analytics view.
- Customized title for the unavailable analytics view on each card.

## Testing steps

- Create a JN site with Woo but no Jetpack.
- Disable the Analytics feature in Woo > Settings > Advanced > Features.
- Log in to the store on the app with an application password and confirm that the analytics unavailable view is displayed on all 4 cards: performance, top performers, stock, and most active coupon cards.
- Repeat the same steps for a Jetpack/WPCom store to confirm the same behavior.

## Screenshots


Performance | Top Performers | Coupons | Stock
--- | --- | --- | ---
<img src="https://github.com/user-attachments/assets/5d9241c2-bfc5-4895-b141-b3960b805f7e" width=320 /> | <img src="https://github.com/user-attachments/assets/4831d345-8644-4adf-befd-35a0c92d1867" width=320 /> | <img src="https://github.com/user-attachments/assets/805e4728-55bb-4d81-ac6b-b2f6622177fa" width=320 /> | <img src="https://github.com/user-attachments/assets/6b4e469c-4f25-4357-8533-c770e7b39143" width=320 />


---
